tests/components/wmspro/test_cover.py

```diff
@@ -0,0 +1,223 @@
+"""Test the wmspro diagnostics."""
+
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.components.wmspro.const import DOMAIN, MANUFACTURER
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def test_cover_device(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_awning: AsyncMock,
+) -> None:
+    """Test that a cover device is created correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) == 2
+
+    device_registry = dr.async_get(hass)
+    device_entry = device_registry.async_get_device(identifiers={(DOMAIN, "58717")})
+    assert device_entry is not None
+    assert device_entry.manufacturer == MANUFACTURER
+    assert device_entry.name == "Markise"
+    assert device_entry.serial_number == 58717
```

Comment:
Consider using snapshots to serialize these data structures. `airgradient` has examples of this in `test_init.py`. You can then run `pytest ./tests/components/wmspro --snapshot-update` and it will fixate the data objects in .ambr files

Reply:
I read about this, but the developer documentation stated that snapshot-based tests do not replace unittests.

Reply:
Yes it doesn't replace unit tests that check if something returns something else after a change. But this use case is perfect

Reply:
Thanks for the snapshot improvement.